src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunSymbolNodeFactory.cs

```diff
@@ -814,5 +821,52 @@ public override int GetHashCode()
             }
             return node;
         }
+
+        Dictionary<int, ObjectNode> _rvaFieldSymbols = new Dictionary<int, ObjectNode>();
+
+        public ObjectNode GetRvaFieldNode(FieldDesc fieldDesc)
+        {
+            Debug.Assert(fieldDesc.HasRva);
+            EcmaField ecmaField = (EcmaField)fieldDesc;
+
+            if (!_codegenNodeFactory.TypeSystemContext.InputFilePaths.ContainsKey(ecmaField.Module.Assembly.GetName().Name))
```

Comment:
Can we make this go through the compilation group APIs?

[Compilation group](https://github.com/dotnet/corert/blob/master/Documentation/botr/ilc-architecture.md#dependency-expansion-process) abstracts exactly this kind of checks. We have this API as a choke point so that each time we add a new compilation mode we don't have to duplicate new if checks in dozens of places like this.